exporter/metrics_exporter.go

```diff
@@ -143,13 +143,22 @@ func (m *metricsExporter) recordMetrics() {
 		// all the running workers.
 		if phase == core.WorkerPhaseRunning {
 			var pendingJobs int
-			// TODO: This logic doesn't currently account for the possibility that
-			// there are more pages of events with running workers, so this count
-			// of pending jobs could be wrong.
-			for _, event := range events.Items {
-				for _, job := range event.Worker.Jobs {
-					if job.Status.Phase == core.JobPhasePending {
-						pendingJobs++
+			for events.Continue != "" {
+				events, err = m.apiClient.Core().Events().List(
+					context.Background(),
+					&core.EventsSelector{
+						WorkerPhases: []core.WorkerPhase{phase},
+					},
+					&meta.ListOptions{Continue: events.Continue},
+				)
+				if err != nil {
+					log.Println(err) // TODO: Is this the best way to deal with errors?
+				}
+				for _, event := range events.Items {
+					for _, job := range event.Worker.Jobs {
+						if job.Status.Phase == core.JobPhasePending {
+							pendingJobs++
+						}
```

Comment:
You're moving on to the second page before dealing with the items on the first page. If there is no second page, you're always going to count zero pending jobs.